File: src/Application Specific/Classes/Entity Collections/Quotes/QuotesSQL.cs
using System.Data;

namespace FSM.Entity
{
    namespace Quotes
    {
        public class QuotesSQL
        {
            private Sys.Database _database;

            public QuotesSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public DataView QuotesGet_All()
            {
                _database.ClearParameter();
                var dt = _database.ExecuteSP_DataTable("Quotes_GetAll");
                dt.TableName = Sys.Enums.TableNames.tblQuotes.ToString();
                return new DataView(dt);
            }

            public DataView Quotes_Search(string Criteria)
            {
                _database.ClearParameter();
                _database.AddParameter("@Criteria", Criteria, true);
                var dt = _database.ExecuteSP_DataTable("Quotes_Search");
                dt.TableName = Sys.Enums.TableNames.tblQuotes.ToString();
                return new DataView(dt);
            }

            public DataView QuotesGet_All_For_CustomerID(int CustomerID)
            {
                _database.ClearParameter();
                _database.AddParameter("@CustomerID", CustomerID);
                var dt = _database.ExecuteSP_DataTable("QuotesGet_All_For_CustomerID");
                dt.TableName = Sys.Enums.TableNames.tblQuotes.ToString();
                return new DataView(dt);
            }

            public DataView QuotesGet_All_For_SiteID(int SiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@SiteID", SiteID);
                var dt = _database.ExecuteSP_DataTable("QuotesGet_All_For_SiteID");
                dt.TableName = Sys.Enums.TableNames.tblQuotes.ToString();
                return new DataView(dt);
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Quotes/QuotesSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace Quotes

        Public Class QuotesSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Function QuotesGet_All() As DataView
                _database.ClearParameter()
                Dim dt As DataTable = _database.ExecuteSP_DataTable("Quotes_GetAll")
                dt.TableName = Entity.Sys.Enums.TableNames.tblQuotes.ToString
                Return New DataView(dt)
            End Function

            Public Function Quotes_Search(ByVal Criteria As String) As DataView
                _database.ClearParameter()
                _database.AddParameter("@Criteria", Criteria, True)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("Quotes_Search")
                dt.TableName = Entity.Sys.Enums.TableNames.tblQuotes.ToString
                Return New DataView(dt)
            End Function

            Public Function QuotesGet_All_For_CustomerID(ByVal CustomerID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@CustomerID", CustomerID)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("QuotesGet_All_For_CustomerID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblQuotes.ToString
                Return New DataView(dt)
            End Function

            Public Function QuotesGet_All_For_SiteID(ByVal SiteID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@SiteID", SiteID)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("QuotesGet_All_For_SiteID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblQuotes.ToString
                Return New DataView(dt)
            End Function

#End Region

        End Class

    End Namespace

End Namespace


